### PR TITLE
Prevent citizens from eating glass bottles

### LINKED ISSUE #7038: Beekeeper eating honey bottle also eats the bottle itself.
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: minecolonies-0.14.97-ALPHA-universal.jar
- Structurize Version: structurize-0.13.163-ALPHA-universal.jar
- Related Mods and their Versions: FTB Direwolf20 1.16 modpack version 1.9.0

## Expected behavior
When the colonist eats a honey bottle, an empty bottle will be left in their inventory.

## Actual behavior
When the colonist eats a honey bottle it eats both the honey and the bottle.

## Steps to reproduce the issue

1. Give a colonist some honey bottles (and only that).
2. Wait till they get hungry and eat the honey bottle.
3. Check their inventory to see whether a bottle was left behind.

### Logs

- latest.log: 
- crashlog: 

### Notes
I noticed this issue on a beekeeper (as they eat the honey they harvest). 

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by chandler05**
Closes #7038
Closes #
Closes #

# Changes proposed in this pull request:
- Stop a citizen from eating glass bottles at the same time as honey, as that is very unhealthy. Applies to all items with a specified container item. The bottle/container item will be dropped on the ground if the citizen has no room in their inventory.
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIEatTask.java [hunk 1]
```diff
@@ -257,11 +256,30 @@ private EatingState eat()
...
         final Food itemFood = stack.getItem().getFood();
 
+        final boolean isBottled = stack.getItem() instanceof HoneyBottleItem;
+
         final double satIncrease = itemFood.getHealing() * (1.0 + citizen.getCitizenColonyHandler().getColony().getResearchManager().getResearchEffects().getEffectStrength(SATURATION));
 
         citizenData.increaseSaturation(satIncrease / 2.0);
         citizenData.getInventory().extractItem(foodSlot, 1, false);
 
+        if (isBottled) {
```
Comment (ravenbuilder934) [inline src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIEatTask.java]: Bracket on new line

**Review by ravenbuilder934 — COMMENTED** [review]
Could it maybe depend on the item's containerItem?

**Comment by chandler05** [comment]
I don't know if the vanilla items actually have those values.

**Comment by ravenbuilder934** [comment]
They do, I just checked with honey bottles. (Although in mojmaps it's called itemRemainder or so)

**Comment by chandler05** [comment]
Ok yeah you're right, added.

**Comment by ravenbuilder934** [comment]
would it be possible to drop other items if the containerItem is different (if that makes any sense)? But as it is it definitely solves the initial issue, 👍

**Comment by chandler05** [comment]
So if the item has a container item at all, than make that item drop/go into the inventory as well?

**Comment by ravenbuilder934** [comment]
Yeah

**Review by ravenbuilder934 — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIEatTask.java [hunk 2]
```diff
@@ -257,11 +256,31 @@ private EatingState eat()
...
         final Food itemFood = stack.getItem().getFood();
 
+        final Item containerItem = stack.getItem().getContainerItem();
+
         final double satIncrease = itemFood.getHealing() * (1.0 + citizen.getCitizenColonyHandler().getColony().getResearchManager().getResearchEffects().getEffectStrength(SATURATION));
 
         citizenData.increaseSaturation(satIncrease / 2.0);
         citizenData.getInventory().extractItem(foodSlot, 1, false);
 
+        if (containerItem != null)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIEatTask.java]: might double check if it's air too, just to be sure =D

Comment (chandler05) [reply src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIEatTask.java]: Done 👍

**Comment by Raycoms** [comment]
:shipit:

**Review by ravenbuilder934 — APPROVED** [review]

**Review by Mekle001 — APPROVED** [review]

**Pull request merged**